src/main/java/marquez/common/models/UrnType.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.common.models;
+
+public enum UrnType {
+  DATASET,
+  DATASOURCE;
+
+  @Override
+  public String toString() {
+    return name().toLowerCase();
+  }
+}
```

Comment:
Discussed in-person, but I think it makes sense to add the number of parts as part of the definition of this enum. For example, we'll never have a DATASOURCE URN with 5 parts. It'll have to be exactly 2. So let's just codify that and disallow people to create invalid scenarios that will error out down the line.

Reply:
Additionally, we should have a few unit tests, where we exercise `valueOf` and convert back and forth. It'll remove issues with capitalization, etc.

Reply:
We decided to remove this type in favor of defining the namespace (internally) as part of the class